dev-support/test-patch.sh

```diff
@@ -654,13 +656,64 @@ function compute_gitdiff
   if [[ ! -f ${GITDIFFLINES} ]]; then
     touch "${GITDIFFLINES}"
   fi
+
   if [[ ! -f ${GITDIFFCONTENT} ]]; then
     touch "${GITDIFFCONTENT}"
   fi
 
+  if [[ -s "${GITDIFFLINES}" ]]; then
+    compute_unidiff
+  fi
+
   popd >/dev/null
 }
 
+## @description generate an index of unified diff lines vs. modified/added lines
+## @description ${GITDIFFLINES} must exist.
+## @audience    private
+## @stability   stable
+## @replaceable no
+function compute_unidiff
+{
+  declare fn
+  declare filen
+  declare tmpfile="${PATCH_DIR}/tmp.$$.${RANDOM}"
+
+  # now that we know what lines are where, we can deal
+  # with github's pain-in-the-butt API. It requires
+  # that the client provides the line number of the
+  # unified diff on a per file basis.
+
+  # First, build a per-file unified diff, pulling
+  # out the 'extra' lines, grabbing the adds with
+  # the line number in the diff file along the way,
+  # finally rewriting the line so that it is in
+  # './filename:diff line:content' format
+
+  for fn in ${CHANGED_FILES}; do
+    filen=${fn##./}
+
+    ${GIT} diff ${filen} \
+      | tail -n +6 \
+      | ${GREP} -n '^+' \
+      | ${GREP} -vE '^[0-9]*:\+\+\+' \
+      | ${SED} -e 's,^\([0-9]*:\)\+,\1,g' \
+        -e s,^,./${filen}:,g \
+            >>  "${tmpfile}"
+  done
+
+  # at this point, tmpfile should be in the same format
+  # as gitdiffcontent, just with different line numbers.
+  # let's do a merge (using gitdifflines because it's easier)
+
+  # ./filename:real number:diff number
+  paste -d: "${GITDIFFLINES}" "${tmpfile}" \
+    | ${AWK} -F: '{print $1":"$2":"$5":"$6}' \
```

Comment:
18: note: Expressions don't expand in single quotes, use double quotes for that. [SC2016]